conda/models/records.py

```diff
@@ -390,19 +414,48 @@ def is_extracted(self):
 
     @property
     def tarball_basename(self):
-        return basename(self.package_tarball_full_path)
+        return basename(self.preferred_package_path)
+
+    @property
+    def package_conda_full_path(self):
+        return self.package_tarball_full_path.replace('.tar.bz2', '.conda')
+
+    @property
+    def preferred_package_path(self):
+        return (self.package_conda_full_path if isfile(self.package_conda_full_path) else
+                self.package_tarball_full_path)
 
     def _calculate_md5sum(self):
-        memoized_md5 = getattr(self, '_memoized_md5', None)
-        if memoized_md5:
-            return memoized_md5
+        memo_key = '_memoized_md5'
+        memoized_hash = getattr(self, memo_key, None)
+        if memoized_hash:
+            return memoized_hash
 
         from os.path import isfile
         if isfile(self.package_tarball_full_path):
             from ..gateways.disk.read import compute_md5sum
-            md5sum = compute_md5sum(self.package_tarball_full_path)
```

Comment:
I don't think the `from os.path import isfile` above is needed anymore since it's imported at the beginning